packages/@react-spectrum/list/src/listview.css

```diff
@@ -359,3 +433,20 @@
     padding-top: var(--spectrum-global-dimension-size-50);
   }
 }
+
+.react-spectrum-ListViewItem-checkbox--enter {
+  opacity: 0.01;
+  max-width: 0px;
```

Comment:
max width is a way to animate from 0px width to an unknown width, so long as the max-width is always big enough for the entire content to fit into
80px should be big enough in all settings for a checkbox
we don't want the max-width to be too big though, because then when the frames are divided, the width will change too drastically and it'll look just like the checkbox appeared instantly at the right size

I do need to update this value to use calc from checkbox if people generally like it